server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2089,6 +2090,26 @@ public InputStream maybeRateLimitSnapshots(InputStream stream) {
         return maybeRateLimit(stream, () -> snapshotRateLimiter, snapshotRateLimitingTimeInNanos);
     }
 
+    @Override
+    public CompletableFuture<IndexShardSnapshotStatus> getShardSnapshotStatus(SnapshotId snapshotId,
```

Comment:
Would be useful to mention in the commit message that we are partially reverting https://github.com/crate/crate/commit/cda5f69d1358d78bed29b0288543156544a3936e by bringing back  `getShardSnapshotStatus`

Reply:
Done